### ISSUE TITLE
projects/core/howto/components.rst should be properly formatted

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7641]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7641|
|Type|enhancement|
|Created|2014-09-16 12:03:57Z|
|Branch|https://github.com/twisted/twisted/tree/cleanup-components-rst-7641-2|

Cleaning up rST formatting and such.

Attachments:

* [cleanup-components-rst-7641.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/729/7293cd97da460f33ac5eb2d979e897818a2cc96f/d7e55829b8b261a2d55395ccf7e3f9bf8737ad8d.diff) (27121 bytes) - added by herrwolfe on 2014-09-16 22:45:52Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__7641 7641
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_cleanup_components_rst_7641_2 branches/cleanup-components-rst-7641-2
branch_author__hawkowl__rwall hawkowl, rwall
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1410869037048010 1410869037048010
changetime__1411396650868943 1411396650868943
version__None None
owner__rwall rwall

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [43106]) Branching to cleanup-components-rst-7641.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl removed owner|
|-|-|

Cleaned up whitespace and formatting, and altered one docstring to match the coding standard while I was there.

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_3"></a>|@derwolfe set owner to @hawkowl|
|-|-|

Hi hawkowl,

Thanks for the patch. Everything except for one single line looks good - the line contains two sentences instead of a single sentence per line. I've uploaded a patch, cleanup-components-rst-7641.diff, containing the necessary change (it is against trunk and not the current branch). 

I've also pushed the change to https://github.com/derwolfe/twisted/commit/389c2955c95dfdeef96bceb9a237bac0749ae2d2. It might be easier to see the change using the github link and not the diff.

Once this line is fixed, please merge.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_4"></a>|@wallrj commented|
|-|-|

(In [43142]) Branching to cleanup-components-rst-7641-2.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

I applied herrwolfe's updated patch to a new branch and forced a documentation build. It passed.
 * https://buildbot.twistedmatrix.com/builders/documentation/builds/4870
(Other build slaves are currently broken for other reasons.)

The rendered documentation looks fine.

I'll merge.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_6"></a>|@wallrj commented|
|-|-|

PS Thanks to TimeBandit for helping me review the changes during the PyconUK sprint.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_7"></a>|@wallrj set status to `closed`|
|-|-|

(In [43146]) Merge cleanup-components-rst-7641-2

Author: hawkowl
Reviewers: herrwolfe, TimeBandit
Fixes: [#7641](https://github.com/twisted/twisted/issues/7641)

Reformatted components documentation with semantic sentences and cleaned up white space.
